classification2.Rmd

```diff
@@ -142,10 +149,31 @@ a train / test split of 75% / 25%, as desired. Recall from Chapter \@ref(classif
 that we use the `glimpse` function to view data with a large number of columns,
 as it prints the data such that the columns go down the page (instead of across).
 
+```{r 06-train-prop, echo = FALSE}
+train_prop <- cancer_train |> 
+  group_by(Class) |>
+  summarize(proportion = n()/nrow(cancer_train))
+```
+
+We can use `group_by` and `summarize` to find the percentage of malignant and benign classes 
+in `cancer_train` and we see about `r round(filter(train_prop, Class == "B")$proportion, 2)*100`% of the training
+data are benign and `r round(filter(train_prop, Class == "M")$proportion, 2)*100`% 
+are malignant indicating that our class proportions were roughly preserved when we split the data.
+
+```{r 06-train-proportion}
+cancer_proportions <- cancer_train %>%
+                      group_by(Class) %>%
+                      summarize(n = n()) %>%
+                      mutate(percent = 100*n/nrow(cancer_train))
+cancer_proportions
+```
+
+
+
 ### Preprocess the data
 
-As we mentioned last chapter, KNN is sensitive to the scale of the predictors,
-and so we should perform some preprocessing to standardize them. An
+As we mentioned in the last chapter, $K$-nearest neighbors is sensitive to the scale of the predictors,
+so we should perform some preprocessing to standardize them. An
 additional consideration we need to take when doing this is that we should
 create the standardization preprocessor using **only the training data**. This ensures that
 our test data does not influence any aspect of our model training. Once we have
```

Comment:
below here (github didn't let me comment)

"makes it simple to handle this properly"

I think this was likely something I added -- probably should avoid "simple" in case this is tough for some readers!